src/lava/magma/runtime/message_infrastructure/close_on_shutdown_smm.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: LGPL 2.1 or later
+# See: https://spdx.org/licenses/
+
+
+from multiprocessing.managers import SharedMemoryManager
+
+
+class CloseOnShutdownSMM:
```

Comment:
What do you think about renaming this to `SharedMemoryManager`? The fact that it closes on shutdown is not too important for the user to worry about, I think. It is rather strange that multiprocessing does not do that for you.
For the code below, you could then use
```python
import multiprocessing as mp
self._smm = mp.managers.SharedMemoryManager()
```